docs/content/docs/concepts/glossary.md

```diff
@@ -25,182 +25,605 @@ under the License.
+
+#### Snapshot
+
+A generic term referring to a global, consistent image of the state of a [Flink job](#(flink)-job). 
+A snapshot can be full or incremental and includes a pointer into each of the data sources as well as 
+a copy of the state from each of the job’s stateful operators that resulted from having processed all 
+the events up to those positions in the sources.
+
+Flink periodically takes persistent snapshots of all the state in every operator and copies these 
+snapshots somewhere more durable, such as a distributed file system.
+
+Flink uses a variant of the Chandy-Lamport algorithm known as asynchronous barrier snapshotting.
+
+#### Source
+
+This is the source of the data that gets piped into a [Flink application](#(flink)-application) to be 
+processed. As long as data keeps flowing in, Flink can keep performing calculations. 
+
+A few basic data sources are built into Flink and are always available, such as reading from files, 
+directories, sockets, and ingesting data from collections and iterators. 
 
-#### Flink Session Cluster
+#### Spilling
 
-A long-running [Flink Cluster](#flink-cluster) which accepts multiple [Flink Jobs](#flink-job) for
-execution. The lifetime of this Flink Cluster is not bound to the lifetime of any Flink Job.
-Formerly, a Flink Session Cluster was also known as a Flink Cluster in *session mode*. Compare to
-[Flink Application Cluster](#flink-application-cluster).
+This is a technique where state data is spilled to disk before JVM heap memory is exhausted.
 
 #### State Backend
 
-For stream processing programs, the State Backend of a [Flink Job](#flink-job) determines how its
-[state](#managed-state) is stored on each TaskManager (Java Heap of TaskManager or (embedded)
-RocksDB).
+For stream processing programs, the state backend of a [Flink job](#(flink)-job) determines how its
+[state](#managed-state) is stored on each [TaskManager](#taskmanager).
 
-#### Sub-Task
+Two implementations of state backends are available. One is based on RocksDB, an embedded key/value 
+store that keeps its working state on disk, and the other is heap-based that keeps its working state 
+in memory, on the Java heap.
 
-A Sub-Task is a [Task](#task) responsible for processing a [partition](#partition) of
-the data stream. The term "Sub-Task" emphasizes that there are multiple parallel Tasks for the same
-[Operator](#operator) or [Operator Chain](#operator-chain).
+#### Stream Barriers
+
+A core element of Flink's distributed snapshotting. Stream barriers are injected into the [DataStream](#datastream) 
+and flow with the [records](#record) as part of the DataStream. Barriers never overtake records and 
+flow strictly in line. A barrier separates the records in the DataStream into the set of records that 
+goes into the current snapshot, and the records that go into the next snapshot.
+
+#### Subtask
+
+A subtask is a [task](#task) responsible for processing a [partition](#partition) of the [DataStream](#datastream). 
```

Comment:
```suggestion
A subtask is a logical concept. It represents a [task](#task) responsible for processing a [partition](#partition) of the [DataStream](#datastream). 
```

https://stackoverflow.com/questions/53610342/difference-between-job-task-and-subtask-in-flink